### PR TITLE
Selected branches coverage

### PR DESCRIPTION
**Pull request by bakhtin**
Signed-off-by: Artyom Bakhtin <a@bakhtin.net>

### Description of the Change
Jenkinsfile:
- Run code coverage routine (sonarqube, lcov) for Develop and Master branches only as well as for all PRs. Earlier, code coverage run for commits to either branch.

### Benefits
Significantly reduces CI build time by not executing a long-running code coverage stage.

### Possible Drawbacks 
Code coverage history will not be available for branches other than Develop and Master after this PR will be merged

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -71,7 +71,13 @@ pipeline {
           steps {
             script {
               debugBuild = load ".jenkinsci/debug-build.groovy"
-              debugBuild.doDebugBuild(true)
+              // run coverage if branch is either develop or master, or if it is a PR
+              if (env.BRANCH_NAME in ['develop', 'master'] || env.CHANGE_ID != null) {
+                debugBuild.doDebugBuild(true)
```
[the remark this answers is not in the document]
Comment (muratovv) [reply Jenkinsfile]: Is it true that `env.CHANGE_ID != null` means that only first commit in the branch will be built.

File: Jenkinsfile [hunk 2]
```diff
@@ -157,7 +163,7 @@ pipeline {
             script {
               def coverageEnabled = false
               def cmakeOptions = ""
-              if (!params.Linux) {
+              if (!params.Linux && (env.BRANCH_NAME in ['develop', 'master'] || env.CHANGE_ID != null)) {
```
Comment (muratovv) [inline Jenkinsfile]: also the statement `(env.BRANCH_NAME in ['develop', 'master'] || env.CHANGE_ID != null)` is duplicated maybe move it into own method?

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tyvision) [reply Jenkinsfile]: @muratovv if `CHANGE_ID` is not null, then it is a PR. If `CHANGE_ID` is null, then it is a branch commit

File: Jenkinsfile (hunk 2 above)
Comment (tyvision) [reply Jenkinsfile]: might be the point. will be easier to work with it later

Comment (muratovv) [reply Jenkinsfile]: Ok, we have to review the scripts in several sprints and refactor it.

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (bakhtin) [reply Jenkinsfile]: This only means we should not run coverage if branch is not a PR

[the remark this answers is not in the document]
Comment (bakhtin) [reply Jenkinsfile]: > debugBuild.doDebugBuild(coverageEnabled: true)

Groovy supports named parameters in "it's own way". You can pass them as long as you define function as 
```
def f(Map args) {}

f(a:'par1', b:'par2')
```
Parameters can be later accessed within a function as `args.a or args.b`. The problem is that accepted parameters can not be easily derived from signature of the function.

**Pull request merged**